sycl/test-e2e/Basic/max_work_group_query.cpp

```diff
@@ -0,0 +1,48 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple  %s -o %t.out
+// REQUIRES: cuda || hip || level_zero
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+#include <sycl/sycl.hpp>
+
+#include <cassert>
+#include <iostream>
+
+using namespace sycl;
+
+int main() {
+  queue q;
+  device dev = q.get_device();
+
+// Check if this experimental feature is supported
+#ifdef SYCL_EXT_ONEAPI_MAX_WORK_GROUP_QUERY
```

Comment:
This comment is confusing. I think we should rather check like this:

```c++
#if !defined(SYCL_EXT_ONEAPI_MAX_WORK_GROUP_QUERY)
  #error SYCL_EXT_ONEAPI_MAX_WORK_GROUP_QUERY is not defined!
#endif
```
I think we claim that DPC++ supports this extension, so this macro **must be defined**. Skipping API calls when extension macro is undefined makes sense if portability across multiple SYCL implementations is required. Here we test DPC++ implementation, which must support this extension.